src/ValueObject/FileValueObject.php

```diff
@@ -112,127 +78,47 @@ public static function fromArray(array $values = []): ValueObjectInterface {
     );
 
     if (isset($values['title'])) {
-      $file->setTitle($values['title']);
+      $file = $file->withTitle($values['title']);
     }
 
     if (isset($values['language_code'])) {
-      $file->setLanguageCode($values['language_code']);
+      $file = $file->withLanguageCode($values['language_code']);
     }
 
     return $file;
   }
 
   /**
-   * Getter.
-   *
-   * @return string
-   *   Property value.
-   */
-  public function getUrl(): string {
-    return $this->url;
-  }
-
-  /**
-   * Getter.
-   *
-   * @return string
-   *   Property value.
-   */
-  public function getMime(): string {
-    return $this->mime;
-  }
-
-  /**
-   * Getter.
-   *
-   * @return string
-   *   Property value.
-   */
-  public function getSize(): string {
-    return $this->size;
-  }
-
-  /**
-   * Getter.
-   *
-   * @return string
-   *   Property value.
-   */
-  public function getName(): string {
-    return $this->name;
-  }
-
-  /**
-   * Getter.
-   *
-   * @return string
-   *   Property value.
-   */
-  public function getTitle(): string {
-    return $this->title ? $this->title : $this->name;
-  }
-
-  /**
-   * Getter.
-   *
-   * @return string
-   *   Property value.
-   */
-  public function getLanguageCode(): string {
-    return $this->languageCode;
-  }
-
-  /**
-   * Getter.
-   *
-   * @return string
-   *   Property value.
-   */
-  public function getExtension(): string {
-    return pathinfo($this->name, PATHINFO_EXTENSION);
-  }
-
-  /**
-   * Setter.
+   * Create a new FileValueObject with a specific title.
    *
    * @param string $title
-   *   Property value.
+   *   Title value.
    *
-   * @return $this
+   * @return \Drupal\oe_theme\ValueObject\FileValueObject
+   *   A new FileValueObject.
    */
-  public function setTitle(string $title): FileValueObject {
-    $this->title = $title;
+  public function withTitle(string $title): FileValueObject {
+    $clone = clone $this;
 
-    return $this;
+    $clone->storage['title'] = $title;
+
+    return $clone;
   }
 
   /**
    * Setter.
```

Comment:
Comment not consistent with other analogical methods.

Reply:
Fixed.